astropy_helpers/openmp_helpers.py

```diff
@@ -39,69 +41,222 @@
 }
 """
 
+def _get_flag_value_from_var(flag, var, delim=' '):
+    """
+    Utility to extract `flag` value from `os.environ[`var`]` or, if not present,
+    from `distutils.sysconfig.get_config_var(`var`)`.
+    E.g. to get include path: _get_flag_value_from_var('-I', 'CFLAGS')
+    might return "/usr/local/include".
 
-def add_openmp_flags_if_available(extension):
+    Notes
+    -----
+    Not yet tested and therefore not yet supported on Windows
     """
-    Add OpenMP compilation flags, if available (if not a warning will be
-    printed to the console and no flags will be added)
 
-    Returns `True` if the flags were added, `False` otherwise.
+    if sys.platform.startswith('win'):
+        return None
+
+    # Simple input validation
+    if not var or not flag:
+        return None
+    l = len(flag)
+    if not l:
+        return None
+
+    # Look for var in os.eviron
+    try:
+        flags = os.environ[var]
+    except KeyError:
+        flags = None
+    
+    # If os.environ was unsuccesful try in sysconfig
+    if not flags:
+        try:
+            flags = get_config_var(var)
+        except KeyError:
+            return None
+
+    # Extract flag from {var:value}
+    if flags:
+        for item in flags.split(delim):
+            if flag in item[:l]:
+                return item[l:]
+
+    return None
+
+def _get_include_path():
+    return _get_flag_value_from_var('-I', 'CFLAGS')
+
+def _get_library_path():
+    return _get_flag_value_from_var('-L', 'LDFLAGS')
+
+def get_openmp_flags():
     """
+    Utility for returning compiler and linker flags possibly needed for
+    OpenMP support.
 
-    ccompiler = new_compiler()
-    customize_compiler(ccompiler)
+    Returns
+    -------
+    result : `{'compiler_flags':<flags>, 'linker_flags':<flags>}`
 
-    tmp_dir = tempfile.mkdtemp()
+    Notes
+    -----
+    The flags returned are not tested for validity, use
+    `test_openmp_support(openmp_flags=get_openmp_flags())` to do so.
+    """
 
-    start_dir = os.path.abspath('.')
+    compile_flags = []
+    include_path = _get_include_path()
+    if include_path:
+        compile_flags.append('-I' + include_path)
+
+    link_flags = []
+    lib_path = _get_library_path()
+    if lib_path:
+        link_flags.append('-L' + lib_path)
 
     if get_compiler_option() == 'msvc':
-        compile_flag = '-openmp'
-        link_flag = ''
+        compile_flags.append('-openmp')
     else:
-        compile_flag = '-fopenmp'
-        link_flag = '-fopenmp'
+        compile_flags.append('-fopenmp')
+        link_flags.append('-fopenmp')
+        if lib_path:
+            link_flags.append('-Wl,-rpath,' + lib_path)
 
-    try:
+    return {'compiler_flags':compile_flags, 'linker_flags':link_flags}
+
+def test_openmp_support(openmp_flags=None, silent=False):
```

Comment:
rename this function to `check_openmp_support` or something, it's very confusing to see a test function in the main codebase (and I'm not 100% certain that pytest likes it either).

Reply:
Good point, I hadn't thought of the pytest 'test_' namespace clash, cheers.